pkg/clusteragent/clusterchecks/leadership_kube.go

```diff
@@ -19,5 +19,7 @@ func getLeaderIPCallback() (types.LeaderIPCallback, error) {
 		return nil, err
 	}
 
-	return engine.GetLeaderIP, engine.EnsureLeaderElectionRuns()
+	engine.StartLeaderElectionRun()
```

Comment:
why are we calling this twice? Is there a race between this call and the other?

Reply:
do you mean here and in `EnsureLeaderElectionRuns()`?

with this change `EnsureLeaderElectionRuns` is no longer called on `start()`, but it is called every time the `kube` api server check is run https://github.com/DataDog/datadog-agent/blob/master/pkg/collector/corechecks/cluster/kubernetes_apiserver.go#L215 , which i guess is a periodic way to make sure the leader election is still running.

Reply:
I'm wondering as there's a `sync.Once` which call will effectively be the first to run the function between `getLeaderIPCallback` & `EnsureLeaderElectionRuns` and if both are necessary?

Reply:
ah, so in the case that `GetLeaderEngine()` fails, we would retry inside the method but we will never hit line 22. but  i suppose if the kube apiserver check is running (which it would be if we're inside `getLeaderIPCallback`) it isn't strictly necessary here; do you think it's better to remove it?

Reply:
Ok got it, one is called via `setupClusterCheck` and this other one via the kube apiserver check, I guess the kube apiserver check could be disabled as cluster checks, so we probably need to keep both 👍

Reply:
ok thanks!